### PR TITLE
ENG--9078, change the conflict resolution API.

### PR DESCRIPTION
**Pull request by luyangco**
Change-Id: I09b6b5d47ef66c25b4176e7062275ccf26e229c5

### CODE REVIEW

File: src/frontend/org/voltdb/PartitionDRGateway.java [hunk 1]
```diff
@@ -59,6 +59,26 @@ public static DRRecordType valueOf(int ordinal) {
...
+        public static final ImmutableMap<Integer, DRConflictType> conversion;
+        static {
+            ImmutableMap.Builder<Integer, DRConflictType> b = ImmutableMap.builder();
+            for (DRConflictType t : DRConflictType.values()) {
+                b.put(t.ordinal(), t);
+            }
+            conversion = b.build();
+        }
+
+        public static DRConflictType valueOf(int ordinal) {
```
Comment (nshi) [inline src/frontend/org/voltdb/PartitionDRGateway.java]: You don't need to build the conversion map and write this method. You can just use `DRConflictType.values()[ordinal]` to get the type.

Comment (luyangco) [inline src/frontend/org/voltdb/PartitionDRGateway.java]: Should `DRRecordType` also use `DRRecordType.values()[ordinal]` to get the type at PartitionDRGateway.java:180?

Comment (nshi) [inline src/frontend/org/voltdb/PartitionDRGateway.java]: Yes.

Comment (luyangco) [inline src/frontend/org/voltdb/PartitionDRGateway.java]: Fixed.

**Pull request merged**